SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriverRPC.h

```diff
@@ -0,0 +1,241 @@
+ * which is responsible for swapping the entity the RPC is written on and the counterpart.
+ */
+struct FRPCCrossServerPayload : FRPCPayload
+{
+	uint64 UUID;
+	Worker_EntityId Counterpart;
+};
+
+/**
+ * Additional data accompanying a received RPC.
+ */
+struct FRPCMetaData
+{
+	uint64 Timestamp;
+	FSpatialGDKSpanId SpanId;
+
+	void ComputeSpanId(FName Name, SpatialGDK::SpatialEventTracer& Tracer, SpatialGDK::EntityComponentId EntityComponent, uint64 RPCId);
+};
+
+/**
+ * Wrapper used to add Timestamp and event tracing information to a received RPC.
+ */
+template <typename T>
+struct TimestampAndETWrapper
+{
+	TimestampAndETWrapper(FName InRPCName, Worker_ComponentId InComponentId, SpatialGDK::SpatialEventTracer* InEventTracer = nullptr)
+		: RPCName(InRPCName)
+		, ComponentId(InComponentId)
+		, EventTracer(InEventTracer)
+	{
+	}
+
+	using AdditionalData = FRPCMetaData;
+	struct WrappedData
+	{
+		WrappedData() = default;
+		WrappedData(T&& InData)
+			: Data(MoveTemp(InData))
+		{
+		}
+
+		const T& GetData() const { return Data; }
+
+		const FRPCMetaData& GetAdditionalData() const { return MetaData; }
+
+		T Data;
+		FName ReceiverName;
+		FRPCMetaData MetaData;
+	};
+
+	WrappedData MakeWrappedData(Worker_EntityId EntityId, T&& Data, uint64 RPCId)
+	{
+		WrappedData Wrapper(MoveTemp(Data));
+		Wrapper.ReceiverName = RPCName;
+		Wrapper.MetaData.Timestamp = FPlatformTime::Cycles64();
+		if (EventTracer != nullptr)
+		{
+			Wrapper.MetaData.ComputeSpanId(RPCName, *EventTracer, SpatialGDK::EntityComponentId(EntityId, ComponentId), RPCId);
+		}
+
+		return Wrapper;
+	}
+
+	FName RPCName;
+	Worker_ComponentId ComponentId;
+	SpatialGDK::SpatialEventTracer* EventTracer = nullptr;
+};
+
+class USpatialLatencyTracer;
+
+/**
+ * RPC component for the SpatialNetDriver.
+ * It contains the glue between the RPC primitives (queue, sender, receiver) and Unreal Actors,
+ * with all the methods and callbacks necessary to send and receive RPCs in UnrealEngine.
+ * The base class only contains a receiver for NetMulticast RPCs.
+ * Derived class will contain additional RPC components
+ */
+UCLASS()
+class SPATIALGDK_API USpatialNetDriverRPC : public UObject
+{
+	GENERATED_BODY()
+public:
+	// Definition for a "standard queue", that is, all queued RPC for sending could have an accompanying SpanId.
+	// Makes it easier to define a standard callback for when an outgoing RPC is written to the network.
+	using StandardQueue = SpatialGDK::TWrappedRPCQueue<FSpatialGDKSpanId>;
+
+	USpatialNetDriverRPC();
+	virtual void Init(USpatialNetDriver* InNetDriver, const SpatialGDK::FSubView& InActorAuthSubView,
+					  const SpatialGDK::FSubView& InActorNonAuthSubView);
+
+	void AdvanceView();
+	virtual void ProcessReceivedRPCs();
+	virtual void FlushRPCUpdates();
+	void FlushRPCQueue(StandardQueue& Queue);
+	void FlushRPCQueue(Worker_EntityId, StandardQueue& Queue);
+	TArray<FWorkerComponentData> GetRPCComponentsOnEntityCreation(const Worker_EntityId EntityId);
```

Comment:
I'm not following the singular purpose of this function, looks like it's used for entity creation, as well as flushing rpcs?
Also the overloaded use of it's name and protected variant with a separate signature is confusing, I'd prefer a separate name to help clarify the purpose.